src/connect/ConnectWizard.ts

```diff
@@ -147,32 +89,132 @@ export class ConnectWizard {
+        this.NumberOfSteps = 3;
+        await MultiStepInput.runAsync(async (input) => {
+            await this.prevalidateAsync(ResourceType.Pod, targetResourceName, targetResourceNamespace, currentContext, bridgeClient, kubectlClient, input);
+            let resourceNameToPersist: string = targetResourceName;
+            // Because the specific pod name changes, we persist only the first segments of the name, e.g. <deployment name>-<deployment guid>-.
+            const split: string[] = targetResourceName.split(`-`);
+            if (split.length > 1) {
+                // Delete last segment
+                split.splice(-1, 1);
+                resourceNameToPersist = `${split.join(`-`)}`;
+            }
+
+            this._result.resourceName = resourceNameToPersist;
+            this._result.targetCluster = currentContext.cluster;
+            this._result.targetNamespace = targetResourceNamespace;
+            this._result.resourceType = ResourceType.Pod;
+
+            return await this.getContainerSelection(targetResourceName, ResourceType.Pod, kubectlClient);
+        });
+    }
+
+    private async handleServiceResourceType(targetResourceName: string, targetResourceNamespace: string, currentContext: IKubeconfigEnrichedContext, bridgeClient: any, kubectlClient: any) {
+        await MultiStepInput.runAsync(async (input) => {
+            await this.prevalidateAsync(ResourceType.Service, targetResourceName, targetResourceNamespace, currentContext, bridgeClient, kubectlClient, input);
+            return this.pickServiceAsync(input, currentContext, ResourceType.Service);
         });
-        return this._isWizardComplete ? this._result as IWizardOutput : null;
     }
 
+
+    private async getContainerSelection(targetResourceName: string, targetResourceType: ResourceType, kubectlClient: any): Promise<(input: MultiStepInput) => Promise<any>> {
+        // get the list of containers
+        const containersList = await kubectlClient.getContainerNames(targetResourceName, this._result.targetNamespace);
+
+        if (containersList?.length > 1) {
+            // show containers quick pick list to select the container to debug
+            const containerChoices: vscode.QuickPickItem[] = containersList.map(containers => ({ label: containers }));
+            return (input: MultiStepInput) => this.inputContainersAsync(input, containerChoices, targetResourceType);
+        }
+
+        // single container for the service selected
+        this._result.containerName = containersList?.[0];
+        return (input: MultiStepInput) => this.inputPortsAsync(input, targetResourceType);
+    }
+
+
     private async pickServiceAsync(
         input: MultiStepInput,
         currentContext: IKubeconfigEnrichedContext,
         resourceType: ResourceType
-    ): Promise<(input: MultiStepInput) => Promise<(input: MultiStepInput) => Promise<(input: MultiStepInput) => Promise<void>>>> {
+    ) {
```

Comment:
I'd recommend `export`ing `InputStep` from `MultiStepInput.ts` and adding a return type `Promise<InputStep | void>`. This will allow it to be used in `MultiStepInput.runAsync()` in a type-safe way.